document manageJobsNamespaceSelector (#3748)
kubernetes-sigs/kueue: site/content/en/docs/tasks/manage/enforce_job_management/manage_jobs_without_queue_name.md

---
title: "Setup manageJobsWithoutQueueName"
date: 2024-06-28
weight: 10
description: >
  Using manageJobsWithoutQueueName and mangedJobsNamespaceSelector to prevent admission of Workloads without assigned LocalQueues.
---

This page describes how to configure Kueue to ensure that all Workloads submitted in namespaces

> intended for use by _batch users_ will be managed by Kueue even if they lack a local-queue label.

Reviewer: ```suggestion
intended for use by _batch users_ will be managed by Kueue even if they lack the queue-name label.
```
maybe we also can give a useful link

Author: I changed to use the actual label not a shorthand.

Reviewer: fine

```yaml
matchExpressions:
- key: kubernetes.io/metadata.name
  operator: NotIn
  values: [ kube-system, kueue-system ]
```
This dafault value exempts the `kube-system` and `kueue-system` namespaces from management; all other
namespaces will be managed by Kueue when `manageJobsWithoutQueueName` is true.

Alternatively, the _batch administrator_ can label namespaces that are intended for _batch users_
and define a selector that matches only those namespaces.  For example,
```yaml
managedJobsNamespaceSelector:
  matchLabels:
    kueue.x-k8s.io/managed-namespace: true
```

{{< feature-state state="beta" for_version="v0.10" >}}
{{% alert title="Note" color="primary" %}}

`managedJobsNamespaceSelector` is a Beta feature that is enabled by default.

You can disable it by setting the `ManagedJobsNamespaceSelector` feature gate. Check the [Installation](/docs/installation/#change-the-feature-gates-configuration) guide for details on feature gate configuration.
{{% /alert %}}

## Expected Behavior

In all namespaces that match the namespace selector, any Workloads submitted without a queue-name

> label will be suspended.  These Workloads will not be considered for admission by Kueue unless
> they are edited to have a queue-name label.

Reviewer: ### 
```suggestion
label will be suspended.  These Workloads will not be considered for admission by Kueue until
they are edited to have the queue-name label.
```
nit, wdyt?
